### DISCUSSION TITLE
Customize menu tag helper

### DISCUSSION THREAD

**Discussion by luivis07** [OrchardCMS/OrchardCore#12622] [Q&A] [answered]
I have the following layout for my pages:
![image](https://user-images.githubusercontent.com/4964745/195174480-7687b943-0b3f-469a-a6ab-701bc1e68fd4.png)

Inside my `_Layout.cshtml` I'm doing the following:
`<menu alias="alias:MainMenu" cache-id="MainMenu" cache-fixed-duration="00:05:00" cache-tag="alias:MainMenu"/>`

I have this inside my `_ViewImports.cshtml`:
`@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers`
`@addTagHelper *, OrchardCore.Media`
`@addTagHelper *, OrchardCore.DisplayManagement`
`@addTagHelper *, OrchardCore.ResourceManagement`
`@addTagHelper *, OrchardCore.Contents`
`@addTagHelper *, OrchardCore.Menu`

How can I customize how the menu looks? I saw here: https://github.com/OrchardCMS/OrchardCore/issues/1465 that I should be able to create a `Menu-MainMenu.cshtml` to overwrite how it displays and follow the guidelines here to customize: https://github.com/OrchardCMS/OrchardCore/blob/main/src/docs/reference/modules/Menu/README.md

but every time I do that I get this error: 
C:\Projects\orchardtest\Pages\Index.cshtml(2,22): error CS0103: The name 'Tag' does not exist in the current context [C:\Projects\orchardtest\orchardtest.csproj]
C:\Projects\orchardtest\Pages\Index.cshtml(7,40): error CS0103: The name 'DisplayAsync' does not exist in the current context [C:\Projects\orchardtest\orchardtest.csproj]

**Comment by luivis07** [comment]
So, I've sorted fix this by creating a custom theme and adding the `menu.cshtml` to my theme, etc. (following some of the guidelines here: https://www.youtube.com/playlist?list=PLuskKJW0FhJcSX7j0Bd-1X5hq3dgCtYwO)
However, I also created my own module. So what I want to do know is use my module as the "Default" MVC application for the CMS so I have:
`OrchardWeb.Cms` project (used only for CMS admin)
`Orchard.CustomTheme` (used to control the theme of my app)
`Orchard.MainModule` (want to use it to provide the logic instead of embedding Razor Pages in `OrchardWeb.Cms`

Is it possible to do this? meaning how do I redirect the request to go to my `OrchardCore.MainModule` , i.e. `localhost:0000/` , `localhost:0000/Home` , and `localhost:0000/Home/Index` should go to my 
OrchardCore.MainModule -> HomeController -> Index View

By the way I'm using .Net 6 , not sure if that changes anything.

**Reply by Skrypt** [reply]
Use the Area name on your anchors `asp-area="OrchardCore.MainModule"`.
Else for the home page, there is a site settings configuration in the database where you can change the Area name too.

**Reply by ns8482e** [reply]
Hm... Question is not related to original discussion question, 

If you don't have custom routes setup, default route to any module's controller is ~/Module/Controller/Action

However You can set custom routes for your module in module's startup.cs. Route attributes also work

**Reply by luivis07** [reply]
Sorry if I don't understand fully, but right now this works:
`localhost/MyModule/Controller/Action`
but I would like it to be
`localhost/Controller/Action`
so I want my main module to respond without having to put the module name in the url. 
I believe what @ns8482e  said is to setup custom routes in the MyModule/Program.cs? so I don't need any configuration for routing in my `main.web` CMS app?

**Comment by luivis07** [comment, marked as answer]
I figured it out, found the solution here: https://github.com/OrchardCMS/OrchardCore/discussions/10545

Thank you all!
